File: src/main/java/io/github/oleksivio/telegram/bot/api/model/method/group/SetChatPhoto.java
package io.github.oleksivio.telegram.bot.api.model.method.group;

import io.github.oleksivio.telegram.bot.api.model.NetworkError;
import io.github.oleksivio.telegram.bot.core.controller.network.FileNetworker;
import io.github.oleksivio.telegram.bot.core.model.method.UploadFile;
import io.github.oleksivio.telegram.bot.core.model.responses.BooleanResponse;
import org.springframework.core.io.FileSystemResource;
import io.github.oleksivio.telegram.bot.core.model.responses.CommonResponse;

/**
 * @see <a href="https://core.telegram.org/bots/api#setchatphoto">setChatPhoto</a>
 */
public class SetChatPhoto extends UploadFile<Boolean> {
    private static final String METHOD = "setChatPhoto";

    /**
     * chat_id Integer or String Unique identifier for the target chat or username of the
     * target channel (in the format @channelusername)
     */
    private static final String CHAT_ID_KEY = "chat_id";
    /**
     * photo InputFile New chat photo, uploaded using multipart/form-data
     */
    private static final String PHOTO_KEY = "photo";

    public SetChatPhoto(FileNetworker fileNetworker) {
        super(METHOD, fileNetworker);
    }

    @Override
    protected Class<? extends CommonResponse<Boolean>> getResultWrapperClass() {
        return BooleanResponse.class;
    }

    public SetChatPhoto setPhoto(FileSystemResource photo) {
        putFile(PHOTO_KEY, photo);
        return this;
    }

    public SetChatPhoto setChatId(Long chatId) {
        putLong(CHAT_ID_KEY, chatId);
        return this;
    }

    @Override
    public SetChatPhoto setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/method/group/SetChatPhoto.kt
package io.github.oleksivio.telegram.bot.api.model.method.group

import io.github.oleksivio.telegram.bot.core.model.ApiDict
import io.github.oleksivio.telegram.bot.core.model.BooleanResponse
import io.github.oleksivio.telegram.bot.core.model.method.ActionMap
import io.github.oleksivio.telegram.bot.core.model.method.file.UploadFile
import org.springframework.core.io.FileSystemResource

/**
 * @see [setChatPhoto](https://core.telegram.org/bots/api/#setchatphoto)
 */
data class SetChatPhoto(

        /**
         * chat_id Integer or String Unique identifier for the target chat or username of the
         * target channel (in the format @channelusername)
         */
        val chatId: Long,
        /**
         * photo InputFile New chat photo, uploaded using multipart/form-data
         */
        val photo: FileSystemResource

) : UploadFile<Boolean>() {
    override fun ActionMap.fill() {
        putFile(ApiDict.PHOTO_KEY, photo)
        putLong(ApiDict.CHAT_ID_KEY, chatId)
    }

    override val resultWrapperClass = BooleanResponse::class

    override val method = "setChatPhoto"
}
